tests/e2e/specs/dashboard-samples/RecomendedExtentions.spec.ts

```diff
@@ -109,37 +113,58 @@ suite(`Check if recommended extensions installed for ${samples}`, async function
-            await browserTabsUtil.refreshPage();
-            await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
         });
 
         test(`Check if extensions is installed and enabled`, async function (): Promise<void> {
+            this.retries(10);
             Logger.debug(`ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.`);
             extensionsView = await (await new ActivityBar().getViewControl('Extensions'))?.openView();
 
             Logger.debug(`extensionsView?.getContent().getSections(): get current section.`);
             [extensionSection] = await extensionsView?.getContent().getSections() as ExtensionsViewSection[];
+            await driverHelper.waitAllPresence(webCheCodeLocators.ExtensionsViewSection.itemTitle, TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT);
 
-            Logger.info(`Check if recommendedExtensions.recommendations are installed: ${recommendedExtensions.recommendations}.`);
             for (const extension of recommendedExtensions.recommendations) {
-                Logger.debug(`extensionSection.findItem(${extension}).`);
-                await extensionSection.findItem(extension);
+                Logger.info(`Check if ${JSON.stringify(extension)} are installed.`);
+
+                Logger.debug(`extensionSection.findItem(${extension.name}).`);
+                await extensionSection.findItem(extension.name);
+
+                // check if extension require reload the page
+                if (await driverHelper.isVisible((webCheCodeLocators.ExtensionsViewSection as any).requireReloadButton)) {
+                    Logger.debug(`Extension require reload the editor. Refreshing the page..`);
+                    await browserTabsUtil.refreshPage();
+                    await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+                    Logger.debug(`ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.`);
+                    extensionsView = await (await new ActivityBar().getViewControl('Extensions'))?.openView();
+                    Logger.debug(`extensionsView?.getContent().getSections(): get current section.`);
+                    [extensionSection] = await extensionsView?.getContent().getSections() as ExtensionsViewSection[];
+                    await driverHelper.waitAllPresence(webCheCodeLocators.ExtensionsViewSection.itemTitle, TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT);
+                    Logger.info(`Check if ${JSON.stringify(extension)} are installed.`);
+                    Logger.debug(`extensionSection.findItem(${extension.name}).`);
+                    await extensionSection.findItem(extension.name);
+                }
 
                 Logger.debug(`extensionsView?.getContent().getSections(): switch to marketplace section.`);
                 const [marketplaceSection]: ExtensionsViewSection[] = await extensionsView?.getContent().getSections() as ExtensionsViewSection[];
```

Comment:
I couldn't find UI element with name `Marketplace` in Extension View and in another parts of VS Code Editor.

Could you, please, explain, which particular UI elements `marketplace` and `marketplaceSection` are representing?

Reply:
![image](https://github.com/eclipse/che/assets/72726250/b89a36fc-9485-4284-b482-819fe07b0a4c)
![image](https://github.com/eclipse/che/assets/72726250/4105e3ff-1927-4269-9990-1e12a3142b64)

Reply:
Got it, thanks for explanation!